deploy_controller.rb

```diff
@@ -0,0 +1,63 @@
+require './develop_tag_generator'
+require './test_tag_generator'
+require './git_helper'
+require './user_comms'
+
+class DeployController
+  def initialize(environ, git_helper, user_comms, develop_tag_generator, test_tag_generator)
+    @environ = environ
+    @git_helper = git_helper
+    @user_comms = user_comms
+    @develop_tag_generator = develop_tag_generator
+    @test_tag_generator = test_tag_generator
+  end
+
+  def environment_choice
+    @environ.downcase
+    case @environ
+    when "develop"
+      unless @develop_tag_generator.develop_tag_exists?
+        @user_comms.tell_user_no_develop_tags
+        apply_tag?(@develop_tag_generator.first_tag)
+      else
+        to_increment = increment_choice
+        next_tag = @develop_tag_generator.next_develop_tag(to_increment)
+        apply_tag?(next_tag)
+      end
+
+    when "test"
+      tags_for_this_commit = @git_helper.get_tags_for_this_commit
+      if !@test_tag_generator.check_for_tag?("dev", tags_for_this_commit)
+        @user_comms.tell_user_no_tag("dev")
+      elsif @test_tag_generator.check_for_tag?("test", tags_for_this_commit)
+        @user_comms.tell_user_already_a_tag("test")
+      else
+        apply_tag?(@test_tag_generator.next_tag("dev", "test", tags_for_this_commit))
```

Comment:
in ruby when we use a question mark it means that it returns something that can be used in an if statement. I would say that in this case the best thing to do is to remove the ?